### PR TITLE
upgrade debian12 and bookworm

### PR DESCRIPTION
**Pull request by SagarRawat24**
signed-off-by: Sagar Rawat [sagarrawat24291@gmail.com](mailto:sagarrawat24291@gmail.com)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ARG ARCH=amd64
-FROM --platform=linux/${ARCH} gcr.io/distroless/static-debian11
+FROM --platform=linux/${ARCH} gcr.io/distroless/static-debian12
```
Comment (jmhbnz) [inline Dockerfile:2]: As per the warning in the github issue, this should not be merged yet as https://console.cloud.google.com/gcr/images/distroless/GLOBAL/static-debian12 does not yet exist.

Refer: https://github.com/GoogleContainerTools/distroless/issues/1337

**Review by jmhbnz — COMMENTED** [review]
Thanks for drafting this @SagarRawat24. We will need to hold off merging this until the Distroless image is ready.

File: Dockerfile (hunk 1 above)
Comment (SagarRawat24) [reply Dockerfile:2]: @jmhbnz  is it image does not exist globally or error in code

Comment (jmhbnz) [reply Dockerfile:2]: Hey @SagarRawat24 - The image now exists globally, so these tests should now pass if you re trigger them as follows:

```bash
git reset --soft HEAD~1
git commit -s -a -m "Update to Debian 12 bookworm."
git push --force
```

**Review by jmhbnz — CHANGES_REQUESTED** [review]
Hey @SagarRawat24 - It looks like there is one more file that needs to be updated, can you please check and update `.github/workflows/e2e-arm64-template.yaml`.

**Pull request closed without merging**